OWML.ModLoader/Owo.cs

```diff
@@ -51,28 +52,36 @@ public void LoadMods()
             var priorityMods = _modFinder.GetMods().Where(mod => mod.Manifest.PriorityLoad).ToList();
             var sortedPriority = _sorter.SortMods(priorityMods);
 
-            var modNames = _modFinder.GetMods().Where(mod => mod.Config.Enabled).Select(mod => mod.Manifest.UniqueName).ToList();
+            var modNames = _modFinder.GetMods().Where(mod => mod.Config.Enabled)
+                .Select(mod => mod.Manifest.UniqueName).ToList();
             var sortedMods = sortedPriority.Concat(sortedNormal);
 
-            foreach (var mod in sortedMods)
+            foreach (var modData in sortedMods)
             {
-                foreach (var dependency in mod.Manifest.Dependencies)
+                var isMissingDependency = false;
+                if (modData.Config.Enabled)
                 {
-                    if (!modNames.Contains(dependency) && mod.Config.Enabled)
+                    var missingDependencies = modData.Manifest.Dependencies
+                        .Where(dependency => !modNames.Contains(dependency));
+                    foreach (var dependency in missingDependencies)
                     {
-                        _console.WriteLine($"Error! {mod.Manifest.UniqueName} needs {dependency}, but it's disabled!");
+                        _console.WriteLine($"Error! {modData.Manifest.UniqueName} needs {dependency}, " +
+                                           "but it's disabled/missing!");
+                        isMissingDependency = true;
                     }
                 }
-                var modType = LoadMod(mod);
-                if (modType == null)
+                var modType = LoadMod(modData);
+                if (modType == null || isMissingDependency)
                 {
-                    _logger.Log("Mod type is null, skipping");
-                    _menus.ModsMenu.AddMod(mod, null);
+                    _logger.Log(isMissingDependency ?
+                        "Mod is missing its dependencies, skipping" :
```

Comment:
this missing dependency message would be printed twice (this one and the one above)

Reply:
Yep. How would you want it instead?

Reply:
this only gets printed to log (like when mod is disabled it would print "Mod type is null"). TBH I'm not sure if that `.Log` call is needed at all

Reply:
Right. Stuff's printed in LoadMod and in the dependency check. Can probably be removed 👍 
I don't care enough to do it, tho ;)

Reply:
that first console would also get printed to the log though, and it actually names the mod in it... i don't think this log call is needed

Reply:
removed